youtube_dl/extractor/jwplatform.py

```diff
@@ -50,7 +50,11 @@ def _parse_jwplayer_data(self, jwplayer_data, video_id=None, require_title=True,
             # JWPlayer backward compatibility: flattened sources
             # https://github.com/jwplayer/jwplayer/blob/v7.4.3/src/js/playlist/item.js#L29-L35
             if 'sources' not in video_data:
-                video_data['sources'] = [video_data]
+                if isinstance(video_data, list):
```

Comment:
As it stands, its not going to break other websites. Without this patch, when an array is passed in, a type error exception gets thrown, I believe from L57 where it makes an array of an array.

That said, how would you suggest fixing it in `iqm2.py` without substantial code duplication?